contrib/natss/pkg/dispatcher/dispatcher/dispatcher.go

```diff
@@ -77,10 +83,45 @@ func createReceiverFunction(s *SubscriptionsSupervisor, logger *zap.SugaredLogge
 }
 
 func (s *SubscriptionsSupervisor) Start(stopCh <-chan struct{}) error {
+	// Trigger Connect to establish connection with NATS
+	s.connect <- struct{}{}
 	s.receiver.Start(stopCh)
 	return nil
 }
 
+func (s *SubscriptionsSupervisor) connectWithRetry() {
+	// re-attempting evey 60 seconds until the connection is established.
+	ticker := time.NewTicker(60 * time.Second)
+	defer ticker.Stop()
+	for {
+		nConn, err := stanutil.Connect(clusterchannelprovisioner.ClusterId, clientID, s.natssURL, s.logger.Sugar())
+		if err == nil {
+			s.subscriptionsMux.Lock()
+			s.natssConn = nConn
+			s.subscriptionsMux.Unlock()
+			return
+		}
+		s.logger.Sugar().Errorf("Connect() failed with error: %+v, retrying in 60 seconds", err)
+		select {
+		case <-ticker.C:
+			continue
+		}
+	}
+}
+
+// Connect is called for initial connection as well as after every disconnect
+func (s *SubscriptionsSupervisor) Connect() {
+	for {
+		select {
+		case <-s.connect:
+			// Establishing connection only if there is no already connection to NATS
+			if s.natssConn == nil {
```

Comment:
What sets `s.natssConn` to `nil`? It looks like it will be `nil` when the struct is initialized, but after that, what sets it to `nil`?